docs/developer/api_guidelines.md

```diff
@@ -0,0 +1,438 @@
+| full update     | PUT <resource>             |
+| partial update  | PATCH <resource>           |
+| delete          | DELETE <resource>          |
+
+## list
+Used to search for resources and get back a collection
+* **must** map to HTTP GET method
+* **must not** use a request body
+* **must** return a response body that contains a list of resources that **may** be empty
+* **may** take additional HTTP query parameters
+
+
+## get
+Used for accessing a single resource
+* **must** map to HTTP GET method
+* **must not** use a request body
+* **must** return a response body that contains the resource if successful
+* **may** take additional HTTP query parameters
+
+## create
+Used to create a new resource
+* **must** map to HTTP POST method
+* **may** accept a resource id to choose for the caller. If the resource id already exists, the method **must** fail
+* **may** contain the fields necessary to construct the resource as request body
+* **must** return HTTP 201 - Created on success
+* **should** return the resource id in the *Location* header as a relative URL
+* **should** return a response body that contains the created resource
+
+## full update
+Used to do an update on a resource
+* **must** map to HTTP PUT method
+* **may** be used to create a new resource
+* **must** contain all fields of the resource
+* **must** return HTTP 201 - Created if it was used for creating of a new resource
+* **should** return the resource id in the *Location* header as a relative URL if it was used for creating of a new resource
+* **should** return a response body that contains the resource if it was used for creating of a new resource
+* **must** return an empty response body a HTTP 204 -No content if it was used for updating an existing resource
+
+## partial update
+Used do a partial update on a resource
+* **must** map to HTTP PATCH method
+* **must** contain updated fields in JSON Merge Patch format (https://datatracker.ietf.org/doc/html/rfc7396) in the request body
+* **must** not be used to create a new resource
+* **must** return HTTP 204 -No Content on success
+* **must** return an empty response body
+
+## delete
+Used to remove a resource
+* **must** map to HTTP DELETE method
+* **may** be applicable to a single resource or a collection of resources. A filter criteria **must** be provided as query parameters to delete a collection of resources.
+* **must** not use a request body
+* **must** not use a response body
+* **must** return HTTP 204 - No Content on success
+* **must** return HTTP 202 - Accepted on async. deletion request
+
+---
+# Time Values Conventions
+
+## Time Formats
+
+Time values **should** be represented as UTC times and **must** be encoded according to the [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) standard.
+This standard accepts multiple variations of time formats, however to avoid confustion all time values in our APIs **should** be stored and represented using the following format:
```

Comment:
```suggestion
This standard accepts multiple variations of time formats, however, to avoid confusion all time values in the APIs **MUST** be stored and represented using the following format:
```